test/regridder_tests.jl

```diff
@@ -314,3 +315,45 @@ for FT in (Float32, Float64)
         end
     end
 end
+# test dataset truncation 
+@testset "test dataset truncation" begin
+    # Get the original dataset set up 
+    include(joinpath(pkgdir(ClimaCoupler), "artifacts", "artifact_funcs.jl"))
+    sst_data_all = joinpath(sst_dataset_path(), "sst.nc")
+    ds = NCDataset(sst_data_all, "r")
+    dates = ds["time"][:]
+    first_date = dates[1]
+    last_date = last(dates)
+
+    # values for the truncations 
+    time_start = 0.0
+    time_end = 1.728e6
+    date0test = ["18690101", "18700101", "19790228", "20220301", "20230101"]
```

Comment:
it would be helpful to add a comment explaining which case each of these start dates tests